File: R_Palindrome_Array.py
n = int(input())
arr = list(map(int,input().split()))
temp = arr.copy()

def f(i,j):
    if i >= j:
        if temp == arr:
            return True
        return False
    arr[i],arr[j] = arr[j],arr[i]
    return f(i + 1,j - 1)


print("YES" if f(0,n - 1) else "NO")

File: R_Palindrome_Array.cpp
#include <iostream>
#include <vector>

using namespace std;

bool pali(vector<int>& arr,int i,int j){
    if(i >= j){
        return true;
    }

    if(arr[i] != arr[j]){
        return false;
    }

    return pali(arr,i + 1,j - 1);
}


int main(){
    int n;
    cin>>n;
    vector<int> arr(n);
    
    for(int i = 0;i < n;i++){
        cin>>arr[i];
    }

    if(pali(arr,0,n - 1)){
        cout<<"YES";
    }else{
        cout<<"NO";
    }
    return 0;
}